### PR TITLE
feat: Add local-volume-provisioner to Kubernetes addon

### LINKED ISSUE #922: [ADDON]  local-volume-provisioner
Create addon for https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/tree/master/helm

### PR DESCRIPTION
**Pull request by florentio**
### What does this PR do?
Closes #922

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: modules/kubernetes-addons/local-volume-provisioner/local-static-provisioner/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+name: provisioner
+description: local provisioner chart
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/local-volume-provisioner/local-static-provisioner/Chart.yaml:3]: We should reference the existing chart and not take on re-creating/copying the chart here https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/tree/master/helm/provisioner

**Review by bryantbiggs — CHANGES_REQUESTED** [review]

File: modules/kubernetes-addons/local-volume-provisioner/local-static-provisioner/Chart.yaml (hunk 1 above)
Comment (florentio) [reply modules/kubernetes-addons/local-volume-provisioner/local-static-provisioner/Chart.yaml:3]: Hello @bryantbiggs, there is no helm repo for this chart.
https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/199

I am not sure if setting 
`repository = git+https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner.git@helm?ref=master` will work with the provider 

I will give a try and see the outcome.

**Review by askulkarni2 — APPROVED** [review]
LGTM!

File: modules/kubernetes-addons/local-volume-provisioner/local-static-provisioner/Chart.yaml (hunk 1 above)
Comment (florentio) [reply modules/kubernetes-addons/local-volume-provisioner/local-static-provisioner/Chart.yaml:3]: @bryantbiggs what do you think about merging it as is and when the repo will be available we can switch to it. I do post a comment here https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/issues/289#issuecomment-1240497382 to fix the helm repo for that chart , I will follow up but in the meantime can we keep it like this?

**Pull request merged**